src/components/CityComponent.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, View, Text, ImageBackground } from 'react-native';
+import { utcToZonedTime, format } from 'date-fns-tz';
+import { ru } from 'date-fns/locale';
+import colors from '@styles/colors';
+import { Signs } from '@utils/constants';
+
+export interface ICityComponent {
+    location: string;
+    timeZone: string;
+}
+
+const initialCityData = {
+    weatherType: 'Облачно',
+    temperature: 12,
+    maxTemperature: 12,
+    minTemperature: -3,
+    image: require('@assets/icons/backgroundRain.png'),
+};
+
+export const CityComponent: React.FC<ICityComponent> = props => {
+    const [cityData, setCityData] = useState(initialCityData);
+    const [currentTime, setCurrentDate] = useState(new Date());
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            setCurrentDate(new Date());
+        }, 60000);
```

Comment:
What if current time is 14:40:34 for example, so next time it will update at 14:41:34 and that is 34 seconds later than actual 14:41:00. I think you can check time every second and if currentTime's minutes differ from actual minutes you can setCurrentDate to new date